src/common/pf_alarm.c

```diff
@@ -1815,6 +1811,93 @@ static int pf_alarm_apmr_a_data_ind (
    return ret;
 }
 
+/**
+ * @internal
+ * Return the alarm specifier and maintenance status for a specific sub-slot,
+ * by looking at all diagnosis items found for that subslot.
+ *
+ * Also includes the "current diag item" in the analysis.
+ *
+ * Describes diagnosis alarms.
+ *
+ * This corresponds to the "BuildSummarizedAlarmSpecifier" and
+ * "BuildSummarizedMaintenanceStatus" functions mentioned in the standard.
+ *
+ * @param net              InOut: The p-net stack instance
+ * @param p_ar             In:    The AR instance.
+ * @param api_id           In:    The API identifier.
+ * @param slot_nbr         In:    The slot number.
+ * @param subslot_nbr      In:    The sub-slot number.
+ * @param p_diag_item      In:    The current diag item (may be NULL).
+ * @param p_alarm_spec     Out:   The computed alarm specifier.
+ * @param p_maint_status   Out:   The computed maintenance status.
+ * @return  0  if operation succeeded.
+ *          -1 if an error occurred.
+ */
+static int pf_alarm_get_diag_summary (
```

Comment:
Is this modified or just moved?

Reply:
I doublechecked, this operation is just moved. . I have assumed that we do not want function declaration for the static functions.